octokit/search.go

```diff
@@ -2,47 +2,60 @@ package octokit
 
 import (
 	"github.com/jingweno/go-sawyer/hypermedia"
-	"net/url"
 )
 
-var SearchURL = Hyperlink("search{/type}?q={query}{&page,per_page,sort,order}")
+var SearchURITemplate = "search{/type}?q={query}{&page,per_page,sort,order}"
```

Comment:
Since the API defines four separate link relations for Search, we might be getting a little too DRY with the `type` param here. Hopefully one day we can pull these from the [API front door][root] without needing to rely on hardcoded URI templates.

For consistency, let's define all four here as `Hyperlink` please:

```go
var (
	CodeSearchURL       = Hyperlink("/search/code?q={query}{&page,per_page,sort,order}")
	IssueSearchURL      = Hyperlink("/search/issues?q={query}{&page,per_page,sort,order}")
	RepositorySearchURL = Hyperlink("/search/repositories?q={query}{&page,per_page,sort,order}")
	UserSearchURL       = Hyperlink("/search/users?q={query}{&page,per_page,sort,order}")
)
```

[root]: https://api.github.com/